src/pgduckdb_ruleutils.cpp

```diff
@@ -16,4 +28,215 @@ pgduckdb_function_name(Oid function_oid) {
+	}
+
+	appendStringInfo(&buffer, "CREATE SCHEMA IF NOT EXISTS %s; ", quote_identifier(schema_name));
+
+	appendStringInfoString(&buffer, "CREATE ");
+
+	if (relation->rd_rel->relpersistence != RELPERSISTENCE_TEMP) {
+		elog(ERROR, "Only TEMP tables are supported in DuckDB");
+	}
+
+	appendStringInfo(&buffer, "TABLE %s (", relation_name);
+
+	if (list_length(RelationGetFKeyList(relation)) > 0) {
+		elog(ERROR, "DuckDB tables do not support foreign keys");
+	}
+
+	List *relation_context = pgduckdb_deparse_context_for(relation_name, relation_oid);
+
+	/*
+	 * Iterate over the table's columns. If a particular column is not dropped
+	 * and is not inherited from another table, print the column's name and
+	 * its formatted type.
+	 */
+	TupleDesc tuple_descriptor = RelationGetDescr(relation);
+	TupleConstr *tuple_constraints = tuple_descriptor->constr;
+	AttrDefault *default_value_list = tuple_constraints ? tuple_constraints->defval : NULL;
+
+	bool first_column_printed = false;
+	AttrNumber default_value_index = 0;
+	for (int i = 0; i < tuple_descriptor->natts; i++) {
+		Form_pg_attribute column = TupleDescAttr(tuple_descriptor, i);
+
+		/*
+		 * We disregard the inherited attributes (i.e., attinhcount > 0) here.
+		 * The reasoning behind this is that Citus implements declarative
+		 * partitioning by creating the partitions first and then sending
+		 * "ALTER TABLE parent_table ATTACH PARTITION .." command. This may
+		 * not play well with regular inherited tables, which isn't a big
+		 * concern from Citus' perspective.
+		 */
+		if (!column->attisdropped) {
+			const char *column_name = NameStr(column->attname);
+
+			const char *column_type_name = format_type_with_typemod(column->atttypid, column->atttypmod);
+
+			if (first_column_printed) {
+				appendStringInfoString(&buffer, ", ");
+			}
+			first_column_printed = true;
+
+			appendStringInfo(&buffer, "%s ", quote_identifier(column_name));
+			appendStringInfoString(&buffer, column_type_name);
+
+			if (column->attcompression) {
+				elog(ERROR, "Column compression is not supported in DuckDB");
+			}
+
+			if (column->attidentity) {
+				elog(ERROR, "Identity columns are not supported in DuckDB");
+			}
+
+			/* if this column has a default value, append the default value */
+			if (column->atthasdef) {
+				Assert(tuple_constraints != NULL);
+				Assert(default_value_list != NULL);
+
+				AttrDefault *default_value = &(default_value_list[default_value_index]);
+				default_value_index++;
+
+				Assert(default_value->adnum == (i + 1));
+				Assert(default_value_index <= tuple_constraints->num_defval);
+
+				/*
+				 * convert expression to node tree, and prepare deparse
+				 * context
+				 */
+				Node *default_node = (Node *)stringToNode(default_value->adbin);
+
+				/* deparse default value string */
+				char *default_string = pgduckdb_deparse_expression(default_node, relation_context, false, false);
+
+				if (column->attgenerated == ATTRIBUTE_GENERATED_STORED) {
```

Comment:
DuckDB does support generated virtual columns, which should not be set as a `DEFAULT`

Reply:
Yeah, I realized that. But postgres does not support VIRTUAL columns. I updated the code and comment to be more clear about that.